src/colorizer/TimeControls.ts

```diff
@@ -44,24 +38,46 @@ export default class TimeControls {
   }
 
   private playTimeSeries(onNewFrameCallback: () => void): void {
-    clearInterval(this.timerId);
+    if (this.currentlyPlaying) {
+      return;
+    }
+
+    this.currentlyPlaying = true;
+
+    // TODO: Fix this function so that it doesn't stop the slider from also operating?
+
+    // `lastFrameNum` is a parameter here because relying on `ColorizeCanvas.getCurrentFrame()` can
+    // lead to race conditions that lead to frames getting loaded more than once.
+    const loadNextFrame = async (lastFrameNum: number): Promise<void> => {
+      if (!this.currentlyPlaying) {
+        return;
+      }
+
+      const startTime = Date.now();
+      const nextFrame = this.wrapFrame(lastFrameNum + 1);
 
-    // TODO: Fix this function so that it doesn't stop the
-    // slider from also operating
-    const loadNextFrame = async (): Promise<void> => {
-      if (this.isDisabled) {
-        // stop render loop if time controls have been disabled.
+      if (nextFrame === lastFrameNum) {
+        // Stop playing for single-frame datasets.
+        // TODO: Disable time bar on the UI for datasets with only one frame.
+        this.currentlyPlaying = false;
         return;
       }
-      const nextFrame = this.wrapFrame(this.canvas.getCurrentFrame() + 1);
 
-      // do the necessary update
+      // do the update
       if (this.setFrameFn) {
-        this.setFrameFn(nextFrame);
+        await this.setFrameFn(nextFrame);
       }
+      const endTime = Date.now();
+      const timeElapsed = endTime - startTime;
       onNewFrameCallback();
+
+      // TODO: Could add some sort of smoothing here to make the playback more consistent.
```

Comment:
there is definitely something sketchy about having awaits in the middle of the redraw/rendering call.  might merit some thinking in a future ticket.